src/rp2_common/hardware_rtc/rtc.c

```diff
@@ -11,9 +11,28 @@
 #include "hardware/resets.h"
 #include "hardware/clocks.h"
 
+#define RANGE_CHECK_YEAR(t)  (t->year >= 0 && t->year <= 4095)
+#define RANGE_CHECK_MONTH(t) (t->month >= 1 && t->month <= 12)
+#define RANGE_CHECK_DAY(t)   (t->day >= 1 && t->day <= 32)
+#define RANGE_CHECK_DOTW(t)  (t->dotw >= 0 && t->dotw <= 6)
+#define RANGE_CHECK_HOUR(t)  (t->hour >= 0 && t->hour <= 23)
+#define RANGE_CHECK_MIN(t)   (t->min >= 0 && t->min <= 59)
+#define RANGE_CHECK_SEC(t)   (t->sec >= 0 && t->sec <= 59)
+
+
 // Set this when setting an alarm
 static rtc_callback_t _callback = NULL;
-static bool _alarm_repeats = false;
+static uint8_t _seconds_increment = 1;
```

Comment:
not sure why this is state; it seems like it should be a macro parameter as the usage on line 161 should always be 1 (not the last value)